Library/Homebrew/cmd/update-report.rb

```diff
@@ -882,17 +887,21 @@ def dump_new_formula_report
   def dump_new_cask_report
     return unless Cask::Caskroom.any_casks_installed?
 
-    casks = select_formula_or_cask(:AC).sort.filter_map do |name|
-      name.split("/").last unless cask_installed?(name)
-    end
+    casks = select_formula_or_cask(:AC).sort.reject { |name| cask_installed?(name) }
     return if casks.blank?
 
     ohai "New Casks"
+    should_display_descriptions = if Homebrew::EnvConfig.no_install_from_api?
+      casks.size <= 100
+    else
+      true
+    end
     casks.each do |cask|
-      if (desc = cask_description(cask))
-        puts "#{cask}: #{desc}"
+      cask_token = T.must(cask.split("/").last)
+      if should_display_descriptions && (desc = cask_description(cask))
+        puts "#{cask_token}: #{desc}"
```

Comment:
It seems that we don't display the full cask token for new casks (i.e., `user/tap/foo` is still displayed as `foo`. The changes here are to preserve the existing behaviour.